packages/framework-integration-tests/integration/end-to-end/subscriptions.integration.ts

```diff
@@ -115,6 +116,80 @@ describe('subscriptions', () => {
       await expect(Promise.all([promisify(observableOne), promisify(observableTwo)])).to.eventually.be.fulfilled
     })
   })
+
+  describe('with filters', () => {
+    let client: DisconnectableApolloClient
+    before(async () => {
+      client = await graphQLClientWithSubscriptions()
+    })
+    after(() => {
+      client.disconnect()
+    })
+
+    it('get a carts with a specific ID', async () => {
+      const cartID = random.uuid()
+
+      const originalSubscriptionsCount = await countSubscriptionsItems()
+      // Let's create two subscriptions to the same read model
+      const observable = cartFilteredSubscription(client, { id: { eq: cartID } })
+      // Call the subscribe function to send the subscription to server
+      observable.subscribe(() => {})
+      // Wait for for the subscriptions to arrive
+      await waitForIt(countSubscriptionsItems, (newCount) => newCount == originalSubscriptionsCount + 1)
+      // Check we receive data when the read model is modified
+      await cartMutation(client, cartID)
+      const result = await promisify(observable)
+      const cart = result.data.CartReadModels
+      expect(cart.id).to.equal(cartID)
+    })
+
+    it('get the carts with an specific product id', async () => {
+      const cartID = random.uuid()
+      const productId = random.uuid()
+
+      const originalSubscriptionsCount = await countSubscriptionsItems()
+      // Let's create two subscriptions to the same read model
+      const observable = cartFilteredSubscription(client, {
+        cartItems: { includes: { productId: productId, quantity: 2 } },
+      })
+      // Call the subscribe function to send the subscription to server
+      observable.subscribe(() => {})
+      // Wait for for the subscriptions to arrive
+      await waitForIt(countSubscriptionsItems, (newCount) => newCount == originalSubscriptionsCount + 1)
+      // Check we receive data when the read model is modified
+      await cartMutation(client, cartID, productId)
+      const result = await promisify(observable)
+      const cart = result.data.CartReadModels
+      expect(cart.id).to.equal(cartID)
+
+      const isWellFiltered = cart.cartItems.some((cartItem: { productId: string }) => cartItem.productId === productId)
+      expect(isWellFiltered).to.be.true
+    })
+
+    it('get the cart filtering by an array of strings', async () => {
+      const cartID = random.uuid()
+      const productId = random.uuid()
+
+      const originalSubscriptionsCount = await countSubscriptionsItems()
+      // Let's create two subscriptions to the same read model
+      const observable = cartFilteredSubscription(client, {
+        cartItemsIds: { includes: productId },
```

Comment:
Whaat? then subscriptions with the new schema are working???

Reply:
Yes! 🕺🏻